src/cadecoder.c

```diff
@@ -3642,7 +3642,7 @@ static int ca_decoder_finalize_child(CaDecoder *d, CaDecoderNode *n, CaDecoderNo
         }
 
         if ((d->feature_flags & CA_FORMAT_WITH_CHATTR) != 0 && child->fd >= 0) {
-                unsigned new_attr, old_attr;
+                int new_attr, old_attr;
 
                 new_attr = ca_feature_flags_to_chattr(read_le64(&child->entry->flags) & d->feature_flags);
```

Comment:
both ext4 and btrfs appear to use "unsigned" actually, not int. I think the definition suggested in that mail thread is wrong too and doesn't match reality...

(even the kernel's user-space self-test uses "unsigned" in tools/testing/selftests/efivarfs/open-unlink.c)